tests/test_ir.py

```diff
@@ -232,16 +233,37 @@ def test_is_structurally_equivalent_incompatible_ir_nodes():
     ctx.register_dialect(Cf)
 
     parser = Parser(ctx, program_func)
-    program: ModuleOp = parser.parse_op()
+    program: Operation = parser.parse_op()
 
     assert program.is_structurally_equivalent(program.regions[0]) == False
     assert program.is_structurally_equivalent(
         program.regions[0].blocks[0]) == False
     assert program.regions[0].is_structurally_equivalent(program) == False
     assert program.regions[0].blocks[0].is_structurally_equivalent(
         program) == False
-    assert program.ops[0].regions[0].blocks[0].ops[
-        0].is_structurally_equivalent(
-            program.ops[0].regions[0].blocks[0].ops[1]) == False
-    assert program.ops[0].regions[0].blocks[0].is_structurally_equivalent(
-        program.ops[0].regions[0].blocks[1]) == False
+    if isinstance(program, ModuleOp):
+        assert program.ops[0].regions[0].blocks[0].ops[
+            0].is_structurally_equivalent(
+                program.ops[0].regions[0].blocks[0].ops[1]) == False
+        assert program.ops[0].regions[0].blocks[0].is_structurally_equivalent(
+            program.ops[0].regions[0].blocks[1]) == False
+
+
+@pytest.mark.parametrize("program_str", [(program_region), (program_region_2),
```

Comment:
I would probably create a new file to test `immutable_ir`, so our files are well separated.
I would probably create a new folder, or a new `immutable_ir` file.

Reply:
I did think so myself, but opted to include it here so as to not duplicate all of the IR strings in this file. If you prefer this in a separate file with duplicated code (or can we import them maybe?), I can do that

Reply:
Oh I see, I didn't realize at the moment that we need the same files.
I'm fine to either keep this as it is, or create a new file with the examples, and then separating the files.